src/DynamoCore/Extensions/ReadyParams.cs

```diff
@@ -104,6 +107,36 @@ private void OnCurrentWorkspaceModelCleared(IWorkspaceModel ws)
                 CurrentWorkspaceCleared(ws);
         }
 
+        /// <summary>
+        /// Occurs when current workspace is clearing
+        /// </summary>
+        public event Action<IWorkspaceModel> CurrentWorkspaceClearingStarted;
+        private void OnCurrentWorkspaceModelClearingStarted(IWorkspaceModel ws)
+        {
+            if (CurrentWorkspaceClearingStarted != null)
```

Comment:
in newer c# versions we can now do this:
`CurrentWorkspaceClearingStarted?.Invoke(ws);`